app/backend/lib/gis-upload.ts

```diff
@@ -0,0 +1,106 @@
+import { Router } from 'express';
+import formidable from 'formidable';
+import Ajv from 'ajv';
+import fs from 'fs'; 
+import RateLimit from 'express-rate-limit';
+import schema from './gis-schema.json'; 
+import { performQuery } from './graphql';
+import getAuthRole from '../../utils/getAuthRole';
+
+const limiter = RateLimit({
+  windowMs: 1*60*1000, // 1 minute
+  max: 5
+});
+
+const saveGisDataMutation = `
+  mutation gisUploadMutation($input: JSON) {
+    createGisData(input: {gisData: {jsonData:$input}}) {
+      clientMutationId
+    }
+  }
+`;
+
+const gisUpload = Router();
+const ajv = new Ajv({ allErrors: true, removeAdditional: true });
+
+const formatAjv = (data, errors) => {
```

Comment:
Can you add some comments on this function? There are a few magic numbers. 

Also thoughts on adding types here? 
```suggestion
const formatAjv = (data: Record<string, any>, errors: ErrorObject[]) => {
```